### PR TITLE
Adding a new doc for service worker reg and disable http caching

### PR DESCRIPTION
**Pull request by gauntface**

### CODE REVIEW

File: src/content/en/tools/workbox/guides/service-worker-checklist.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+your service worker file will consult the HTTP cache, which can cause some
+confusion while developing locally or after publishing a new service worker
+and changes don't seem to be taking affect.
+
+Generally, most developers will want to set the `Cache-Control` header
+to 'no-cache', forcing browsers to always check the *server* for
+a new service worker file.
+
+If you are unsure what your current `Cache-Control` header is for your
+service worker, you can check with CURL:
```
Comment (jeffposnick) [inline src/content/en/tools/workbox/guides/service-worker-checklist.md]: "CURL" => "curl"

Probably a good idea to link to https://curl.haxx.se/ as well.

File: src/content/en/tools/workbox/guides/service-worker-checklist.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+<div class="success">
+  <b>DO</b>: Wait until the `window` load event to register your service worker.
+</div>
+
+## Cache-Control of your service worker file
+
+When you register your service worker file, the browser will make a network
+request for the file. If your server returns a `Cache-Control` header with
+your service worker file, the browser will
```
Comment (jeffposnick) [inline src/content/en/tools/workbox/guides/service-worker-checklist.md]: Only Chrome that will do this, as Firefox 57+ [fixed](https://stackoverflow.com/questions/38843970/service-worker-javascript-update-frequency-every-24-hours/38854905#38854905) this behavior to adhere to the current specification.

I'm not sure how to best capture that nuance... I guess rephrase to "...some browsers will..."

File: src/content/en/tools/workbox/guides/service-worker-checklist.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+This page contains a couple of tips that will ensure your users get the
+best experience from your web app as well as ensuring there aren't any
+hidden suprises while developing your site.
+
+## Register your service worker the right way
+
+In general, you should wait for the window load event before registering your
+service worker. This will allow the browser to prioritise assets for the page
+and will prevent any risk of precaching interferring with the page.
```
Comment (philipwalton) [inline src/content/en/tools/workbox/guides/service-worker-checklist.md]: s/interferring/interfering/

File: src/content/en/tools/workbox/guides/service-worker-checklist.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+<div class="success">
+  <b>DO</b>: Wait until the `window` load event to register your service worker.
+</div>
+
+## Cache-Control of your service worker file
+
+When you register your service worker file, the browser will make a network
+request for the file. If your server returns a `Cache-Control` header with
+your service worker file, some browsers browser will
```
Comment (philipwalton) [inline src/content/en/tools/workbox/guides/service-worker-checklist.md]: Remove unneeded "browser".

File: src/content/en/tools/workbox/guides/service-worker-checklist.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+**cache your service worker file in the HTTP cache**. Future requests for
+your service worker file will consult the HTTP cache, which can cause some
```
Comment (philipwalton) [inline src/content/en/tools/workbox/guides/service-worker-checklist.md]: Hmmm, not sure if "consult" is the right word here. Maybe "check"?

File: src/content/en/tools/workbox/guides/service-worker-checklist.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+confusion while developing locally or after publishing a new service worker
+and changes don't seem to be taking affect.
```
Comment (philipwalton) [inline src/content/en/tools/workbox/guides/service-worker-checklist.md]: s/affect/effect/

**Pull request merged**